### ISSUE TITLE
[Feat][Reduction] Support multi-dim reduction for logsumexp_fwd

### ISSUE THREAD

**Issue by lcy-seso** [tile-ai/TileOPs#795]
## Problem

`logsumexp_fwd` manifest specifies `dim: "int | list[int]"` (matching `torch.logsumexp`), but the current implementation only supports single-int `dim`. This blocks the status flip from `spec-only` to `implemented`.

Verified locally:
```python
# TileOPs — raises TypeError
LogSumExpOp(dtype=torch.float16, dim=[1, 2])(x)

# PyTorch — works, returns reduced tensor
torch.logsumexp(x, dim=[1, 2])
```

## Root cause

The kernel (`logsumexp_fwd.py`) takes `(M, N)` as compile-time parameters, assuming a single reduction axis. Multi-dim reduction requires either:
- Multiple sequential single-dim reductions, or
- A kernel that can flatten multiple dims into one reduction axis

## Acceptance criteria

- [ ] `LogSumExpOp(dtype=..., dim=[1, 2])` produces correct results matching `torch.logsumexp(x, dim=[1, 2])`
- [ ] `LogSumExpOp(dtype=..., dim=[0, 2], keepdim=True)` shape and values match PyTorch
- [ ] Tests added for multi-dim cases
- [ ] Manifest validator passes with `--check-op logsumexp_fwd`
- [ ] Status flipped to `implemented` after all checks pass

## Context

Discovered during PR #794 (softmax family spec migration). The `_SoftmaxBaseOp` spec path already handles single-int dim with `movedim` transpose. Multi-dim requires extending this to flatten multiple dims before the (M, N) reshape.

Note: `softmax_fwd` and `log_softmax_fwd` only specify `dim: int` (single), so they are not affected.
